integrations/crowdnode/src/main/java/org/dash/wallet/integrations/crowdnode/api/CrowdNodeBlockchainApi.kt

```diff
@@ -51,7 +52,8 @@ open class CrowdNodeBlockchainApi @Inject constructor(
 
     suspend fun topUpAddress(accountAddress: Address, amount: Coin, emptyWallet: Boolean = false): Transaction {
         val topUpTx = paymentService.sendCoins(accountAddress, amount, null, emptyWallet)
-        return walletData.observeTransactions(true, LockedTransaction(topUpTx.txId)).first()
+        topUpTx.waitToMatchFilters(LockedTransaction())
+        return topUpTx
```

Comment:
Somewhat different pattern here: if we already have a transaction, observe itself. If we don't observe the wallet. 
The rest of CoinJoin transactions don't need this because we're waiting for response and not for them to become locked.